org.eclipse.buildship.core/src/main/java/org/eclipse/buildship/core/internal/DefaultGradleBuild.java

```diff
@@ -130,19 +132,28 @@ public GradleConnectionOperation(Function<ProjectConnection, ? extends T> action
 
         @Override
         public void runInToolingApi(CancellationTokenSource tokenSource, IProgressMonitor monitor) throws Exception {
-            // TODO (donat) use proxy for ProjectConnection to preconfigure inputs, progress logging, etc for all long-running operations
-            // TODO (donat) use proxy for long-running operations to auto-close connections (see ConnectionAwareLauncherProxy)
             // TODO (donat) use AutoCloseable once we update to Tooling API 5.0
-            GradleConnector connector = GradleConnector.newConnector();
-            DefaultGradleBuild.this.gradleBuild.getBuildConfig().toGradleArguments().applyTo(connector);
-            ProjectConnection connection = connector.connect();
+            ProjectConnection connection = IdeAttachedProjectConnection.newInstance(tokenSource, getGradleArguments(), monitor);
             try {
                 this.result = this.action.apply(connection);
             } finally {
                 connection.close();
             }
         }
 
+        private  GradleArguments getGradleArguments() {
+            org.eclipse.buildship.core.internal.configuration.BuildConfiguration config = DefaultGradleBuild.this.gradleBuild.getBuildConfig();
+            return GradleArguments.from(
```

Comment:
❓ Why doesn't this call `config.toGradleArguments()`?

Reply:
Fixed